### PR TITLE
US020 - Delete a role 

### PR DESCRIPTION
**Pull request by Breanna-loughrey**
Screenshot 2023-12-14 at 12 03 31

- Added delete-job-role endpoint and deleteJobRole in JobRoleService and JobRoleDao, as well as getJobRole.

Screenshot 2023-12-14 at 12 08 32

Testing has also been added for this PR:
Screenshot 2023-12-18 at 14 02 02

### CODE REVIEW

**Review by lullabelle — CHANGES_REQUESTED** [review]
Hi
We need a member of the team to review before Josh or I can!

**Review by autumncraig — APPROVED** [review]

File: FlashFeetAPI/src/main/java/org/example/api/JobRoleService.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+public class JobRoleService {
+
+    private JobRoleDao jobRoleDao = new JobRoleDao();
+
+    public JobRoleService(){
+
+    }
+
+    public JobRoleService(JobRoleDao jobRoleDao, DatabaseConnector databaseConnector) {
+    }
```
Comment (jkelso662) [inline FlashFeetAPI/src/main/java/org/example/api/JobRoleService.java]: Is there a reason you need these empty constructors? Especially the second one that ignores the two parameters?

File: FlashFeetAPI/src/main/java/org/example/cli/JobRole.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package org.example.cli;
+
+public class JobRole {
+
+    public JobRole(){
+
+    }
+
+    private String jobRole;
```
Comment (jkelso662) [inline FlashFeetAPI/src/main/java/org/example/cli/JobRole.java]: When we create a job role, should we not use the constructor to pass in the parameters and set them?

File: FlashFeetAPI/src/main/java/org/example/trueApplication.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    @Override
+    public String getName() {
+        return "true";
+    }
+
+    @Override
+    public void initialize(final Bootstrap<trueConfiguration> bootstrap) {
+        // TODO: application initialization
```
Comment (jkelso662) [inline FlashFeetAPI/src/main/java/org/example/trueApplication.java]: Can probably remove this as you have already implemented

**Review by jkelso662 — CHANGES_REQUESTED** [review]
I think there are a few things to do with constructors in this but other than that - looking good! Nearly there :) Again I'm seeing these gitignore files eta in so many of your teams PRs! Can't wait to get them merged haha

File: FlashFeetAPI/src/main/java/org/example/exception/DatabaseConnectionException.java
Comment (jkelso662) [inline FlashFeetAPI/src/main/java/org/example/exception/DatabaseConnectionException.java:1]: Does this need implimented?

File: FlashFeetAPI/src/test/java/org/example/api/service/JobRoleServiceTests.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+                Mockito.when(jobRoleDao.deleteJobRole(jobRoleString)).thenReturn(1);
+                assertEquals(1, jobRoleService.deleteJobRole(jobRoleString, databaseConnector));
+        }
+}
+
+
+
+
+
+
```
Comment (jkelso662) [inline FlashFeetAPI/src/test/java/org/example/api/service/JobRoleServiceTests.java:69]: Watch out for all this blank space! haha Take pride in making your classes neat and tidy 😄

File: FlashFeetAPI/src/main/java/org/example/db/JobRoleDao.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class JobRoleDao {
+    private DatabaseConnector databaseConnector = new DatabaseConnector();
+    public List<JobRole> getAllJobRoles(Connection connection) throws SQLException {
+
+        // Connection c = databaseConnector.getConnection();
```
Comment (jkelso662) [inline FlashFeetAPI/src/main/java/org/example/db/JobRoleDao.java]: 🚨

File: FlashFeetAPI/.gitignore
Comment (jkelso662) [inline FlashFeetAPI/.gitignore:1]: I've seen this file sooo many times! hahaha

File: .idea/FlashFeetAPI.iml
Comment (jkelso662) [inline .idea/FlashFeetAPI.iml:1]: These files will need removed from your repo as they should be ignored now by your gitignore file

**Review by jkelso662 — CHANGES_REQUESTED** [review]
Couple of small things to fix but nearly there! Great PR title and description - I'd love to see the other PR titles match this format.

**Comment by lullabelle** [comment]
your checks are still failing here but we would love to get this reviewed for you before we go? Can you fix?

**Comment by CDawson8** [comment]
That is the workflow fixed so all checks are now passing. Forgot I would need to fix the workflow file on each individual branch, but should be passing the checks now.

**Pull request closed without merging**